src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelAPITests.cs

```diff
@@ -16,6 +16,60 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class SemanticTests : CSharpTestBase
     {
+        [Fact]
+        public void UnamangedConstraintOnExtensionMethod()
+        {
+            var src = @"
+public static class FooExtensions
+{
+    public static T GenericExtension<T>(ref this T self) where T : unmanaged
+    {
+        return self;
+    }
+}
+";
+            var src2 = @"
+public class Program
+{
+    public struct Foo<T> where T : unmanaged
+    {
+        public T t;
+    }
+
+    static void UseFoo<T>(Foo<T> foo) where T : unmanaged
+    {
+        foo. GenericExtension();
+    }
+}";
+            var comp1 = CreateCompilation(src, parseOptions: TestOptions.Regular8);
+
+            var comp2 = CreateCompilation(src2, parseOptions: TestOptions.Regular8,
+                references: new[] { comp1.ToMetadataReference() });
+            comp2.VerifyDiagnostics();
+            Assert.NotNull(checkSymbolInfo(comp2).Symbol);
+
+            comp2 = CreateCompilation(src2, parseOptions: TestOptions.Regular7_3,
+                references: new[] { comp1.ToMetadataReference() });
+            comp2.VerifyDiagnostics(
+                    // (11,14): error CS8652: The feature 'unmanaged constructed types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //         foo. GenericExtension();
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "GenericExtension").WithArguments("unmanaged constructed types").WithLocation(11, 14));
+            var info = checkSymbolInfo(comp2);
```

Comment:
Does this test contain the scenario where `currentCompilation` is null due to the public APIs called by the semantic model? Sorry that I'm a little unclear on when/why this can happen.

Reply:
Yup, the chain is GetSymbolInfo -> try to get reduced extension method -> check constraints -> boom

Reply:
The problem is http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/MethodSymbol.cs,684 where we just assume you can reduce extension methods without a compilation. I mean, you can, but you may not get full constraint checks if you do.

Reply:
Note: public API: http://source.roslyn.io/#Microsoft.CodeAnalysis/Symbols/IMethodSymbol.cs,fe1886144a64d90d,references

Reply:
I thought about just skipping constraint checking entirely when `currentCompilation` is null but that seemed even more risky.

Reply:
OK, just to be crystal clear: do both calls to `checkSymbolInfo` wind up creating a ReducedExtensionMethodSymbol with a `null` Compilation?

Is this an issue where there's not necessarily a "correct" compilation we can associate with the ReducedExtensionMethodSymbol?

Reply:
Yup, both calls pass null as the compilation. The problem is that reducing extension methods may involve constraint checking, which we don't necessarily have a compilation to do that in

Reply:
Is that true any time we construct a generic method with constraints on the type parameters?